packages/vscode-extension/src/network/providers/NetworkProvider.tsx

```diff
@@ -26,6 +27,59 @@ interface NetworkProviderProps extends NetworkTracker {
   toggleTimelineVisible: () => void;
   fetchAndOpenResponseInEditor: (networkLog: NetworkLog) => Promise<void>;
   getResponseBody: (networkLog: NetworkLog) => Promise<ResponseBodyData | undefined>;
+  getThemeData: (themeDescriptor: ThemeDescriptor) => Promise<ThemeData>;
+}
+
+function responseBodyListener(
+  resolve: (value: ResponseBodyData | undefined) => void,
+  messageId: string,
+  requestId: string,
+  responseBodiesRef: React.RefObject<Record<string, ResponseBodyData | undefined>>
+) {
+  const listener = (message: MessageEvent) => {
+    try {
+      const { command, payload }: WebviewMessage = message.data;
+      if (command !== WebviewCommand.CDPCall || payload.id !== messageId) {
+        return;
+      }
+
+      const bodyData = payload.result as ResponseBodyData | undefined;
+
+      if (bodyData === undefined) {
+        resolve(responseBodiesRef.current[requestId]);
+        window.removeEventListener("message", listener);
```

Comment:
Since this function is responsible for calling `window.removeEventListener`, I think it'd be better if it also registered the listener.

Reply:
Same goes for `themeListener`